### PR TITLE
feat(datetype): datetype array

### PR DESCRIPTION
**Pull request by b41sh**
support array datatype

```sql
create table test (v array(int64));
insert into test values([1, 2, 3]), ([4, 5, 6]);
```

### CODE REVIEW

File: src/ast/data_type.rs [hunk 1]
```diff
@@ -140,7 +140,10 @@ impl fmt::Display for DataType {
             DataType::Text => write!(f, "TEXT"),
             DataType::String => write!(f, "STRING"),
             DataType::Bytea => write!(f, "BYTEA"),
-            DataType::Array(ty) => write!(f, "{}[]", ty),
+            DataType::Array(ty) => match **ty {
+                DataType::Text => write!(f, "{}[]", ty),
```
Comment (sundy-li) [inline src/ast/data_type.rs]: Why `Text` case is special?

Comment (b41sh) [reply src/ast/data_type.rs]: This is the syntax of `PostgreSQL`, maybe we don't need to be compatible with it.
https://www.postgresql.org/docs/current/arrays.html

Comment (sundy-li) [reply src/ast/data_type.rs]: Yes, it's fixed size array, we do not support it for now. 

Just `Arary(ty)`

Comment (b41sh) [reply src/ast/data_type.rs]: ok

**Pull request merged**